src/Core/Grid/Query/EmailLogsQueryBuilder.php

```diff
@@ -85,16 +85,14 @@ private function getQueryBuilder(array $filters)
             }
 
             if ('date_add' === $name) {
-                if (isset($value['from'], $value['to'])) {
-                    $qb->andWhere('m.date_add BETWEEN :date_from AND :date_to');
-                    $qb->setParameter('date_from', $value['from']);
-                    $qb->setParameter('date_to', $value['to']);
-                } elseif (isset($value['from'])) {
+                if (isset($value['from'])) {
                     $qb->andWhere('m.date_add >= :date_from');
-                    $qb->setParameter('date_from', $value['from']);
-                } elseif (isset($value['to'])) {
+                    $qb->setParameter('date_from', sprintf('%s %s', $value['from'], '0:0:0'));
```

Comment:
it's a static string, no need to add it in parameter.

Reply:
well yes, but it causes trouble with parameter binding as it breaks SQL or produces wrong results.

Using this approach: `$qb->andWhere('m.date_add >= :date_from 0:0:0');` it breaks SQL, i guess because `:date_from` becomes `"2018-08-02"` and the end result is `m.date_add >= "2018-08-02" 0:0:0`. But it's just a guess, i didnt investigate it deeper.

And using this way: `$qb->andWhere('m.date_add >= ":date_from 0:0:0"');` does not break SQL, but produces wrong results, i guess the same reason as above.


@PierreRambaud do you have better suggestions on how to implement it in clean way?

Reply:
You should pass a real DateTime instead:

```php
->setParameter('date_from', new \DateTime($value['from']), \Doctrine\DBAL\Types\Type::DATETIME);
```

Reply:
Why do we need to manage that here in the first place? Why can't we have the form data with the values of hours, minutes and seconds already set up?

Reply:
i guess because legacy pages didn't have that? Otherwise, i dont know. :confused: